lib/Sema/ConstraintGraph.cpp

```diff
@@ -242,8 +242,10 @@ void ConstraintGraphNode::notifyReferencingVars(
 
 void ConstraintGraphNode::notifyReferencedVars(
     llvm::function_ref<void(ConstraintGraphNode &)> notification) const {
-  for (auto *fixedBinding : getReferencedVars()) {
-    notification(CG[fixedBinding]);
+  for (auto *referencedVar : getReferencedVars()) {
+    auto *repr = referencedVar->getImpl().getRepresentative(/*record=*/nullptr);
+    if (!repr->getImpl().getFixedType(/*record=*/nullptr))
```

Comment:
Since getRepresenative + check for fixed type is now happening in a few places, could we maybe package that into a method to make it easier to update/use in the future?